tests/OSPSuite.Core.Tests/Domain/FormulaTaskSpecs.cs

```diff
@@ -366,6 +366,92 @@ public void should_throw_an_exception()
       }
    }
 
+   internal class When_expanding_all_dynamic_formula_in_a_model_using_in_children_criteria : concern_for_FormulaTask
+   {
+      private IModel _model;
+      private IParameter _sumVolume;
+      private IParameter _volume1_1;
+      private IParameter _volume1_2;
+      private IParameter _volume2_1;
+      private Container _container1;
+      private Container _container2;
+      private Container _subContainer1_1;
+      private Container _subContainer1_2;
+      private Container _subContainer2_1;
+      private Parameter _volume1;
+
+      protected override void Context()
+      {
+         base.Context();
+         A.CallTo(() => _objectBaseFactory.Create<ExplicitFormula>()).Returns(new ExplicitFormula());
+         _model = new Model();
+         _container1 = new Container().WithName("Container1");
+         _container2 = new Container().WithName("Container2");
+         _subContainer1_1 = new Container().WithName("SubContainer1_1").WithParentContainer(_container1);
+         _subContainer1_2 = new Container().WithName("SubContainer1_2").WithParentContainer(_container1);
+         _subContainer2_1 = new Container().WithName("SubContainer2_1").WithParentContainer(_container2);
+
+         _volume1_1 = new Parameter().WithName("volume").WithFormula(new ConstantFormula(1));
+         _volume1_2 = new Parameter().WithName("volume").WithFormula(new ConstantFormula(2));
+         _volume2_1 = new Parameter().WithName("volume").WithFormula(new ConstantFormula(3));
+         _volume1 = new Parameter().WithName("volume").WithFormula(new ConstantFormula(4));
+         _sumVolume = new Parameter().WithName("dynamic").WithFormula(new ConstantFormula(3));
+
+         //simulate a real ROOT Container
+         var root = new Container().WithContainerType(ContainerType.Simulation).WithName("root");
+         _model.Root = root;
+
+         //creating the following structure
```

Comment:
@PavelBal Please check this out. I think this is what you meant but I am not sure :)

Reply:
I think this is what I meant, lol :D Looks good. @msevestre